### DISCUSSION TITLE
Legion 7 Pro 16IRX8H

### DISCUSSION THREAD

**Discussion by davidjman** [4JX/L5P-Keyboard-RGB#131] [General]
Will the Legion 7 Pro 16IRX8H ('23) ever be supported?  This works great on my Legion 5 Pro 16ITH6 ('22 I think) but running this on the L7 Pro tells me it isn't supported. I know the L7 Pro doesn't use the quad-zone color system like the L5 does, so that may be why? 

Reason I ask: Vantage for the L7 Pro doesn't offer a way to "blend" colors across the keyboard like the L5 (which ONLY lets you do that) so you have to manually color-blend yourself instead, row-by-row or column-by-column (or even key-by-key). The only smooth color blending L7 Vantage offers is full-on rainbow mode, which is a neat "party trick" but very distracting for typing, as are pretty much **all** it's other modes, save for "Always" mode.

**Comment by Labtec343** [comment]
Hi I have Lenovo 5 Pro 16ITH6 '16 but it drops this error, did you have some way to fix it?
<img width="382" alt="Captura de pantalla 2023-10-27 020257" src="https://github.com/4JX/L5P-Keyboard-RGB/assets/102999899/5746c82b-d1c9-469f-9d2d-2a8dd6519c86">

**Reply by Labtec343** [reply]
<img width="677" alt="Captura de pantalla 2023-10-27 021331" src="https://github.com/4JX/L5P-Keyboard-RGB/assets/102999899/462c677c-4f34-4cf7-afa5-aa3de6d4a871">

**Reply by davidjman** [reply]
Did you Run as Administrator?  This app should work on 5i 16ITH6.

**Reply by Labtec343** [reply]
Yes I do, my laptop it´s a 5 Pro 2021-2022 16ITH6

**Reply by davidjman** [reply]
Hmm; same model/year as mine (edit: I have both a 5 and a 7).  Well, a couple things you might try (if you haven't already); (1) "Bouncing" the keyboard device in Device Manager (i.e., removing it and allowing Windows to re-add it);  (2) Ye olde faithful reboot;  (3) Hit up Lenovo Website for keyboard drivers.

Though, based on the error output "1mAccess is denied. (os err 5)," you might have the application stored somewhere that it doesn't have full permissions to run from.  Have you copied (for example) into Program Files or something?

Perhaps a better question, was the program working for you _before?_

**Reply by Labtec343** [reply]
yes I´ve done all of that, and yes I changed the location of the installer and also the same.

No the program is first time, but I think when I update the Bios since I bought the Laptop like a year or two years ago it doesn´t work

**Reply by davidjman** [reply]
I have a '21/'22 16ITH6 as well, with the latest updates, and it works for me, so I'm thinking that there's still something about your user permissions or the installation location that it's not liking; the clue is in the error: "Failed to Save File" and "Access Denied (OS Err 5)."  Seems like its failing to write its first-launch configuration file or something?  I'm assuming you've tried simply extracting it to your desktop and running it as Admin from there?

**Reply by Labtec343** [reply]
ok right now the CMD window doesn´t appears just the first pic with the warning window, can you please tell me what Bios version you have

**Reply by davidjman** [reply]
So I actually have two L5P's, one of them is BIOS version H1CN45WW, and one of them is BIOS version H1CN49WW. Both work just fine with Legion RGB.

**Reply by Labtec343** [reply]
I really do not know what to do and also i don´t know where to find one of those bios version to test, I have the last one H1CN53WW

**Reply by 4JX** [reply]
The "Access is denied" error comes from the program not being able to save its configuration/state file ("failed to save file" on the upper part of the error) though for some reason its weirdly formatted due to missing color support even though I explicitly disabled it outputting colors on that terminal. (The weird arrows with ] and extra characters are just information wrongly interpreted as text).

**Reply by Labtec343** [reply]
Do you know a way to find why " failed to find a valid keyboard" or "Hardware error" issues is causing the program do not open?
